shm/include/ship.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "cargo.hpp"
+#include <algorithm>
+#include <memory>
```

Comment:
Please clean up that includes. Do you really need algorithm in that hpp file?

Reply:
This is the remains before function was transfered to source, thanks for attention.